util.c

```diff
@@ -619,20 +624,7 @@ ruby_each_words(const char *str, void (*func)(const char*, int, void*), void *ar
 static inline bool
 rb_uniform_num_cmp_bool(VALUE a, VALUE b)
 {
-    if ((FIXNUM_P(a)))
-    {
-        if (FIXNUM_P(b))
-            return a < b;
-        else
-            return FIX2LONG(a) < RFLOAT_VALUE(b);
-    }
-    else
-    {
-        if (FIXNUM_P(b))
-            return RFLOAT_VALUE(a) < FIX2LONG(b);
-        else
-            return RFLOAT_VALUE(a) < RFLOAT_VALUE(b);
-    }
+    return OPTIMIZED_CMP(a,b) == -1;
```

Comment:
Wasn’t this specialized comparison significant?

Reply:
You mentioned me. I wrote direct comparison manually in this function. Until I found `OPTIMIZED_CMP`.
Which definition is quite same with manually inline and be more safe. 
`rb_uniform_num_cmp_bool` is no need any more

Reply:
In newest commit, I expand `OPTIMIZED_CMP` in `rb_uniform_num_cmp_bool` and make it fix to compare Fixnum and Float